cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -790,12 +793,14 @@ func TestNumCPUCheck(t *testing.T) {
 	}
 
 	for _, rt := range tests {
-		warnings, errors := NumCPUCheck{NumCPU: rt.numCPU}.Check()
-		if len(warnings) != rt.numWarnings {
-			t.Errorf("expected %d warning(s) but got %d: %q", rt.numWarnings, len(warnings), warnings)
-		}
-		if len(errors) != rt.numErrors {
-			t.Errorf("expected %d warning(s) but got %d: %q", rt.numErrors, len(errors), errors)
-		}
+		t.Run(strconv.Itoa(rt.numCPU), func(t *testing.T) {
```

Comment:
possibly add something else as part of the name here?
`fmt.Sprintf(number of CPUs: %d, rt.numCPU)`

or add proper `name` element as part of the `tests` struct.